src/Components/Payment/SavedCreditCards.tsx

```diff
@@ -0,0 +1,181 @@
+          show={this.state.isErrorModalOpen}
+          contactEmail="support@artsy.net"
+        />
+      </>
+    )
+  }
+
+  onCloseModal = () => {
+    this.setState({ isErrorModalOpen: false })
+  }
+
+  private deleteCreditCard(me, id) {
+    this.setState({ isCommittingMutation: true }, () => {
+      commitMutation<SavedCreditCardsDeleteCreditCardMutation>(
+        this.props.relay.environment,
+        {
+          onCompleted: (data, errors) => {
+            const {
+              deleteCreditCard: { creditCardOrError },
+            } = data
+
+            if (creditCardOrError.creditCard) {
+              this.setState({ isCommittingMutation: false })
+            } else {
+              this.onMutationError(errors)
+            }
+          },
+          onError: this.onMutationError.bind(this),
+          mutation: graphql`
+            mutation SavedCreditCardsDeleteCreditCardMutation(
+              $input: DeleteCreditCardInput!
+            ) {
+              deleteCreditCard(input: $input) {
+                creditCardOrError {
+                  ... on CreditCardMutationSuccess {
+                    creditCard {
+                      id
+                      __id
+                    }
+                  }
+                  ... on CreditCardMutationFailure {
+                    mutationError {
+                      type
+                      message
+                      detail
+                    }
+                  }
+                }
+              }
+            }
+          `,
+          variables: {
+            input: { id },
+          },
+          updater: (store, data) => this.onCreditCardDeleted(store, me, data),
+        }
+      )
+    })
+  }
+
+  private onCreditCardDeleted(store, me, data) {
+    const {
+      deleteCreditCard: { creditCardOrError },
+    } = data
+
+    if (creditCardOrError.creditCard) {
+      const mutationPayload = store.getRootField("deleteCreditCard")
+      const creditCardOrErrorEdge = mutationPayload.getLinkedRecord(
+        "creditCardOrError"
+      )
+      const creditCardEdge = creditCardOrErrorEdge.getLinkedRecord("creditCard")
+      const creditCardId = creditCardEdge.getValue("__id")
+      const meStore = store.get(me.__id)
+      const connection = ConnectionHandler.getConnection(
+        meStore,
+        "UserSettingsPayments_creditCards"
+      )
+      ConnectionHandler.deleteNode(connection, creditCardId)
+    }
+  }
+
+  private onMutationError(errors) {
+    console.error("SavedCreditCards.tsx", errors)
+    this.setState({ isErrorModalOpen: true, isCommittingMutation: false })
+  }
+}
+
+export const RemoveLink = styled.div`
+  text-align: right;
+  &:hover {
+    cursor: pointer;
+  }
+`
+
+const SpinnerContainer = styled.div`
+  padding-right: 30px;
+  position: relative;
+`
+
+export class SavedCreditCards extends React.Component<SavedCreditCardsProps> {
+  constructor(props) {
+    super(props)
+  }
+
+  render() {
+    return (
+      <Theme>
```

Comment:
Will all of these components have a common root? If so maybe can create a Form component and then put `<Theme>` in there?